bin/oneoff/remove_no_ttl_partial_registrations.rb

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env ruby
+
+require 'socket'
+require 'timeout'
+
+require_relative '../../dashboard/config/environment'
+
+# This script inspects all cache entries in the partial registration namespace for the presence of a TTL.
+# If no TTL is present, the cache entry is removed.
+DRY_RUN = false
+
+puts "Processing... DRYRUN=#{DRY_RUN}"
+
+PARTIAL_REGISTRATION_SUFFIXES = [
+  /-partial-sso-migrated$/,
+  /-partial-sso$/,
+  /-partial-email$/
+]
+
+all_keys_without_expiration = []
+
+dalli_data = CDO.shared_cache.instance_variable_get(:@data)
+memcached_endpoints = dalli_data.instance_variable_get(:@servers)
+
+puts "Found endpoints #{memcached_endpoints}"
+
+memcached_endpoints.each do |endpoint|
+  hostname, port = endpoint.split(":")
+
+  puts "Checking #{hostname} #{port}"
+  sock = TCPSocket.new(hostname, 11211)
+
+  begin
+    # metadump doesn't return END in our version of memcached, wait 30 seconds for all output instead
+    # https://github.com/memcached/memcached/issues/667
+    Timeout.timeout(30) do
+      sock.write("lru_crawler metadump all\r\n")
+
+      # read keys until encountered an end (for memcached, this will be the Timeout)
+      loop do
+        data = sock.readline
+        break if !data || data.empty? || data == "END\r\n" || data == "ERROR\r\n"
+        matches = data.scan(/^key=(?<key>.*) exp=-1/)
+        break if matches.empty?
+        all_keys_without_expiration += matches.flatten!
```

Comment:
Based on [the Ruby docs](https://apidock.com/ruby/v2_5_5/Array/%2B), it might improve performance to use `concat` if we end up manipulating a large array.